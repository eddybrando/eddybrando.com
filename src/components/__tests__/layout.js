import React from "react"
import renderer from "react-test-renderer"

import Layout from "../layout"

describe("Layout", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Layout
          children={<div />}
          headerTitle={{
            append: "ipsum",
            highlight: "highlighted text",
            prepend: "Lorem",
          }}
          mainContainerClass="foo-class"
          nav={[
            { label: "First link", link: "/" },
            { label: "Second link", link: "/second-link" },
          ]}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
