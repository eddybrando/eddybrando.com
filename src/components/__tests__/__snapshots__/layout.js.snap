// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="page-grid"
>
  <header
    className="header"
  >
    <nav
      className="nav"
    >
      <div
        className="header-links"
      >
        <a
          className="link-block"
          href="/"
          style={Object {}}
          title="First link"
        >
          First link
           
          <svg
            aria-hidden={true}
            className="link-arrow"
            focusable="false"
            height={16}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "willChange": "transform",
              }
            }
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.3 3.7L13.1 7.5 1 7.5 1 8.5 13.1 8.5 9.3 12.3 10 13 15 8 10 3z"
            />
          </svg>
        </a>
        <a
          className="link-block"
          href="/second-link"
          style={Object {}}
          title="Second link"
        >
          Second link
           
          <svg
            aria-hidden={true}
            className="link-arrow"
            focusable="false"
            height={16}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "willChange": "transform",
              }
            }
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.3 3.7L13.1 7.5 1 7.5 1 8.5 13.1 8.5 9.3 12.3 10 13 15 8 10 3z"
            />
          </svg>
        </a>
      </div>
    </nav>
    <section
      className="react-reveal title"
      style={
        Object {
          "opacity": undefined,
        }
      }
    >
      <h1
        className="headline mb-0"
      >
        Lorem
        <span
          className="highlight"
        >
          highlighted text
        </span>
        ipsum
      </h1>
    </section>
  </header>
  <svg
    className="header-bg-curve"
    viewBox="0 0 1440 320"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0,224L80,197.3C160,171,320,117,480,117.3C640,117,800,171,960,186.7C1120,203,1280,181,1360,170.7L1440,160L1440,0L1360,0C1280,0,1120,0,960,0C800,0,640,0,480,0C320,0,160,0,80,0L0,0Z"
      fill="#bfd4f7"
      fillOpacity="1"
    />
  </svg>
  <main
    className="react-reveal main-container foo-class"
    style={
      Object {
        "opacity": undefined,
      }
    }
  >
    <div />
  </main>
  <div
    className="animated-bg"
  >
    <div
      className="bg-element"
    />
    <div
      className="bg-element"
    />
    <div
      className="bg-element"
    />
    <div
      className="bg-element"
    />
    <div
      className="bg-element"
    />
    <div
      className="bg-element"
    />
    <div
      className="bg-element"
    />
    <div
      className="bg-element"
    />
    <div
      className="bg-element"
    />
    <div
      className="bg-element"
    />
  </div>
</div>
`;
